node/src/reactor/main_reactor.rs

```diff
@@ -394,34 +395,229 @@ impl reactor::Reactor for MainReactor {
             MainEvent::MainReactorRequest(req) => {
                 req.0.respond((self.state, self.last_progress)).ignore()
             }
-            MainEvent::MainReactorAnnouncement(
-                ref ann @ ReactorAnnouncement::CompletedBlock { ref block },
-            ) => {
+            MainEvent::HotBlockAnnouncement(HotBlockAnnouncement(HotBlock {
+                block,
+                execution_results,
+                mut state,
+            })) => {
+                debug!(
+                    "handling hot block {} {} {:?}",
+                    block.height(),
+                    block.hash(),
+                    state
+                );
+                if !state.is_stored() {
+                    return fatal!(
+                        effect_builder,
+                        "block should be stored after execution or accumulation"
+                    )
+                    .ignore();
+                }
+
                 let mut effects = Effects::new();
-                effects.extend(self.dispatch_event(
-                    effect_builder,
-                    rng,
-                    MainEvent::EventStreamServer(event_stream_server::Event::BlockAdded(
-                        block.clone(),
-                    )),
-                ));
-                effects.extend(self.dispatch_event(
-                    effect_builder,
-                    rng,
-                    MainEvent::DeployBuffer(deploy_buffer::Event::Block(block.clone())),
-                ));
-                effects.extend(self.dispatch_event(
+
+                let should_update_switch_block = match self.block_accumulator.local_tip() {
```

Comment:
Nit: `should_update_switch_block` is a bit of a confusing name here, because it being true doesn't mean that we _will_ update the switch block - it just means that we will do the check _whether_ we have to update the switch block and update it if needed.

Reply:
Updated in [e70a836](https://github.com/casper-network/casper-node/pull/3549/commits/e70a836f70a16c5c9508184aa6ebafddfb6418a5).